File: shared/src/main/java/ru/privatenull/pnauth/dialog/DialogBody.java
package ru.privatenull.pnauth.dialog;

import net.kyori.adventure.text.Component;
import java.util.Map;

/** Content element rendered in a native Minecraft dialog body. */
public sealed interface DialogBody permits DialogBody.PlainMessage, DialogBody.Item {
    record PlainMessage(Component content, int width) implements DialogBody { }
    record Item(Map<String, Object> itemStack, Component description, int descriptionWidth,
                boolean showDecorations, boolean showTooltip, int width, int height) implements DialogBody {
        public Item { itemStack = Map.copyOf(itemStack); }
    }
}

File: shared/src/main/kotlin/ru/privatenull/pnauth/dialog/DialogBody.kt
package ru.privatenull.pnauth.dialog

import net.kyori.adventure.text.Component

/** Content element rendered in a native Minecraft dialog body. */
sealed interface DialogBody {
    @JvmRecord
    data class PlainMessage(val content: Component, val width: Int) : DialogBody

    @JvmRecord
    data class Item(
        val itemStack: Map<String, Any>,
        val description: Component,
        val descriptionWidth: Int,
        val showDecorations: Boolean,
        val showTooltip: Boolean,
        val width: Int,
        val height: Int
    ) : DialogBody
}
